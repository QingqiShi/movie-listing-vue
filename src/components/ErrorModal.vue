<template>
  <div class="error-modal__backdrop">
    <div class="error-modal">
      <icon name="error" />
      <span><slot /></span>
    </div>
  </div>
</template>

<script>
import Icon from "@/components/Icon.vue";

export default {
  components: { Icon }
};
</script>

<style lang="scss">
.error-modal {
  width: 40%;
  max-width: 15rem;
  background-color: $error;
  color: $on-error;
  transform: translate(0, -20vh);
  padding: 1rem;
  border-radius: 0.3rem;
  text-align: center;

  &__backdrop {
    position: fixed;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .material-icons {
    display: block;
    font-size: 4rem;
    margin-bottom: 2rem;
  }
}
</style>
