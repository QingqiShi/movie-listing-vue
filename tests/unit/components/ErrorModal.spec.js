import { shallowMount } from "@vue/test-utils";
import ErrorModal from "@/components/ErrorModal.vue";

describe("ErrorModal.vue", () => {
  it("renders", () => {
    const wrapper = shallowMount(ErrorModal, {
      slots: { default: "<div>test</div>" }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
