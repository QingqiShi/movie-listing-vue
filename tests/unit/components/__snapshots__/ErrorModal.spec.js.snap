// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal.vue renders 1`] = `
<div class="error-modal__backdrop">
  <div class="error-modal">
    <icon-stub name="error"></icon-stub> <span><div>test</div></span>
  </div>
</div>
`;
